### DISCUSSION TITLE
support full-text search for logs

### DISCUSSION THREAD

**Discussion by debuger6** [openGemini/openGemini#19] [Ideas]
**Is your feature request related to a problem? Please describe.（如果该特性或者功能和已知某个问题相关，请描述问题是什么）**
It's well known that logs are time series data in essence. Different from common time series data, logs actually need to support full-text search. So we can extend the indexing capability to support logs.

**Describe the solution you'd like（问题的解决办法是什么）**
Add full-text index for the log field to support full-text searching. For the indexing capability and implementation, refer to lucene, a well-known open source search engine in the industry.

**Describe alternatives you've considered（描述您考虑过的替代方案）**
We may even develop our own full-text index, which is richer in features and better performance than lucene.

**Additional context（其他补充内容）**
No more.

**Comment by xiangyu5632** [comment]
Agreed. Corresponding string functions need to be added.
